docs/content/3.cookbook/2.pass-custom-headers-and-cookies-to-requests.md

```diff
@@ -0,0 +1,263 @@
+
+## Steps
+::steps
+#step-1
+**Import Required Modules**
+
+```javascript
+import { initSDK, buildModule } from '@vue-storefront/sdk';
+import { client, boilerplateModule, BoilerplateModuleType } from '../../packages/sdk/src';
+import useHeaders from 'path-to-useHeaders-file'; // Replace with the correct path to the `useHeaders` composable.
+```
+
+#step-2
+**Define the SDK Configuration**
+```javascript
+const sdkConfig = {
+  boilerplate: buildModule<BoilerplateModuleType>(boilerplateModule, {
+    apiUrl: 'http://localhost:8181/boilerplate',
+  }),
+};
+```
+#step-3
+**Fetch Headers with the useHeaders Composable**
+
+Use the `useHeaders` composable to fetch the necessary headers. This composable manages both server-side and client-side headers for you.
+
+```javascript
+const { addHeadersToState, headerData } = useHeaders();
+addHeadersToState();
+```
+
+#step-4
+**Setting Up Axios Interceptor**
+
+Before making requests using axios, establish an interceptor to automatically append the desired headers.
+
+#### a. **Check for Existing Interceptor**
+
+Check if an interceptor already exists and eject (remove) it to prevent shared headers across instances.
+
+```javascript
+let interceptorId: number | null = null;
+
+if (interceptorId !== null) {
+  client.interceptors.request.eject(interceptorId);
+}
+```
+
+#### b. **Define the Interceptor**
+
+Append custom headers to each request using the interceptor.
+
+```javascript
+interceptorId = client.interceptors.request.use(
+  (config) => {
+    if (!config.headers) {
+      config.headers = {};
+    }
+
+    // Add custom headers here
+    config.headers['amazing-header'] = 'coolest header ever';
+    config.headers['dev-mode'] = headerData.value?.isDevMode ?? 'dude';
+
+    return config;
+  },
+  (error) => {
+    // Handle request error here
+    return Promise.reject(error);
+  }
+);
+```
+
+#step-4
+**Initialize and Export the SDK**
+
+Finally, initialize the SDK with your configuration and export it.
+
+```javascript
+return initSDK<typeof sdkConfig>(sdkConfig);
+```
+
+#step-5
+**sdk.config.ts** all together
+
+```javascript
+import { initSDK, buildModule } from '@vue-storefront/sdk';
+import { client, boilerplateModule, BoilerplateModuleType } from '../../packages/sdk/src';
+
+// Maintain a reference to the interceptor
+let interceptorId: number | null = null;
+
+export const useSdk = () => {
+  const sdkConfig = {
+    boilerplate: buildModule<BoilerplateModuleType>(boilerplateModule, {
+      apiUrl: 'http://localhost:8181/boilerplate',
+    }),
+  };
+
+  const {addHeadersToState, headerData} = useHeaders()
+
+  addHeadersToState()
+
+  // If an interceptor is already set, eject it to remove it
+  // this prevents headers being shared across instances
+  if (interceptorId !== null) {
+    client.interceptors.request.eject(interceptorId);
+  }
+
+  interceptorId = client.interceptors.request.use(
```

Comment:
I guess it will become flaky and unpredictable when there are many parallel connections to applications and shared client

Reply:
Could you elaborate? Not sure about the dangers if somebody uses the SDK server-side (e.g. in an Express application) but when it comes to the usage inside a storefront, how can SDK client be shared across multiple users running the storefront in their own browser? 🤔

Reply:
I confirm that's the case.
![image](https://github.com/vuestorefront/vue-storefront/assets/39009379/5fbd4297-8b2a-4ac8-bbf8-c4e8ba84cd85)